parsers/thueringen.py

```diff
@@ -0,0 +1,103 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+from utils import Parser
+from datetime import datetime, timedelta
+from pyopenmensa.feed import LazyBuilder
+
+calendar_week_regex = re.compile('\d{4}-\d{1,2}$')
```

Comment:
Prefer declaring variables close to where they are used. Readability trumps performance, since you can always optimize readable code, when you've determined that the performance is important.

Reply:
I moved the variables to the functions they are used at except one which is used at multiple places